src/ASM/ASMs3DLag.C

```diff
@@ -1054,3 +1054,39 @@ void ASMs3DLag::generateThreadGroups (char lIndex, bool, bool)
   threadGroupsFace[lIndex].calcGroups(d1,d2,1);
   threadGroupsFace[lIndex].applyMap(map);
 }
+
+
+bool ASMs3DLag::getGridParameters (RealArray& prm, int dir, int nSegPerSpan) const
+{
+  if (svol)
+     return this->ASMs3D::getGridParameters(prm, dir, nSegPerSpan);
+
+  if (nSegPerSpan < 1)
+  {
+    std::cerr <<" *** ASMs3DLag::getGridParameters: Too few knot-span points "
+              << nSegPerSpan+1 <<" in direction "<< dir << std::endl;
+    return false;
+  }
+
+  int nel = dir == 0 ? (nx-1)/(p1-1) : (dir == 1 ? (ny-1)/(p2-1) : (nz-1)/(p3-1));
```

Comment:
Prefer rename this variable to nel1 (number of elements in one direction). nel is confused as total number of elements in the patch (member in ASMbase).

Reply:
fixed.